modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3189,7 +3189,7 @@ void GDScriptAnalyzer::reduce_preload(GDScriptParser::PreloadNode *p_preload) {
 			p_preload->resolved_path = parser->script_path.get_base_dir().path_join(p_preload->resolved_path);
 		}
 		p_preload->resolved_path = p_preload->resolved_path.simplify_path();
-		if (!FileAccess::exists(p_preload->resolved_path)) {
+		if (!FileAccess::exists(p_preload->resolved_path) && !FileAccess::exists(p_preload->resolved_path + ".import")) {
```

Comment:
Just use `ResourceLoader::exists` instead of all this.

Reply:
I tested and confirmed that this fixes the issue. I'll push an update directly to this PR to do that change and squash the commits / make the commit message clearer.